prefq_documentation.md

```diff
@@ -0,0 +1,96 @@
+# Set up your own PrefQ-Server
+
+For setting up your own server there's absolutely no previous knowledge about Cloud Hosting, Network Protocols, Flask et cetera required. 
+
+By following this guide, you will be able to host your own server for remote Human Feedback Collection within less than 60 minutes, without the need of spending a single dollar.
+
+This whole project follows a strict seperation of concerns, with 3 different instances.
+1. **Query Client**: sends queries to Server & receives Feedback from Server
+2. **Feedback Client**: receives queries from Server & sends Feedback to Server
+3. **Server**: Receives queries, stores them locally & serves them, whenever requested. Only once all Feedback has been evaluated, returns the Feedback to the Query Client. 
+
+Seperating the Query Client from the Server allows for more flexibility. For instance, you can send Queries to the server from your laptop, then disconnect from the web, and still serve videos to the Feedback Client. Additionally, the Query Client can generate videos and train the feedback model using any desired programming language. Communication happens using an html-request interface. 
+
+## Development Stage Server (local)
+
+For a quick first impression - or if you'd like to adjust the server script to your individual needs - it is not necessary to host a remote server. Instead you can simply run the provided server script on localhost. This behavior is specified within the last lines of `server.py`:
+
+```python
+if __name__ == "__main__":
+    before_first_request()
+    app.run(host="localhost", port=5000, debug=True)
```

Comment:
We should make the host and the port(s) configurable on the command line (using argparse). This is out of scope for this PR, I will create a separate issue.